xain_fl/fl/coordinator/aggregate.py

```diff
@@ -108,53 +100,42 @@ def aggregate(
             aggregation_weights = np.ones_like(aggregation_data) / len(aggregation_data)
 
         # compute weighted average
-        aggregated_model_weights: List[ndarray] = [
-            np.sum(
-                [
-                    model_weight * aggregation_weight
-                    for model_weight, aggregation_weight in zip(
-                        model_weights_per_idx, aggregation_weights
-                    )
-                ],
-                axis=0,
-            )
-            for model_weights_per_idx in zip(*mult_model_weights)
-        ]
+        aggregated_model_weights: ndarray = np.sum(
+            [
+                model_weight * aggregation_weight
+                for model_weight, aggregation_weight in zip(mult_model_weights, aggregation_weights)
```

Comment:
if naming the variable `multiple_model_weights` as suggested above will cause this line to wrap to a new line, and make it less readable, then i would suggest to pass the zipped lists to a separate variable (just so that on this line it's shorter) (but still without using abbreviations), and perhaps separate the list comprehension from the numpy summation. but that's very much a detail. just minding readability only

Reply:
changed.